crates/matrix-sdk/src/client/builder.rs

```diff
@@ -471,19 +472,29 @@ impl ClientBuilder {
         let mut sliding_sync_proxy =
             self.sliding_sync_proxy.as_ref().map(|url| Url::parse(url)).transpose()?;
 
-        #[allow(unused_variables)]
-        if let Some(well_known) = well_known {
-            #[cfg(feature = "experimental-sliding-sync")]
-            if sliding_sync_proxy.is_none() {
-                sliding_sync_proxy =
-                    well_known.sliding_sync_proxy.and_then(|p| Url::parse(&p.url).ok())
+        #[cfg(feature = "experimental-sliding-sync")]
+        if self.is_simplified_sliding_sync_enabled {
+            // When using Simplified MSC3575, don't use a sliding sync proxy, allow the
+            // requests to be sent directly to the homeserver.
+            tracing::info!("Simplified MSC3575 is enabled, ignoring any sliding sync proxy.");
+            sliding_sync_proxy = None;
+        } else {
+            // Otherwise, if a proxy wasn't set, use the one discovered from the well-known.
+            if let Some(well_known) = well_known {
+                if sliding_sync_proxy.is_none() {
+                    sliding_sync_proxy =
+                        well_known.sliding_sync_proxy.and_then(|p| Url::parse(&p.url).ok())
+                }
             }
         }
 
         #[cfg(feature = "experimental-sliding-sync")]
-        if self.requires_sliding_sync && sliding_sync_proxy.is_none() {
-            // In the future we will need to check for native support on the homeserver too.
-            return Err(ClientBuildError::SlidingSyncNotAvailable);
+        if self.requires_sliding_sync {
+            if self.is_simplified_sliding_sync_enabled {
+                // TODO: The server doesn't yet expose the availability of SSS.
+            } else if sliding_sync_proxy.is_none() {
+                return Err(ClientBuildError::SlidingSyncNotAvailable);
+            }
```

Comment:
I would coalesce all these checks together, since it's not clear that we'd keep `requires_sliding_sync` for `is_simplified_sliding_sync` (at least we'd have some method named after it). Instead, we could even throw an error when `requires_sliding_sync` and `is_simplified_sliding_sync` are set at the same time, because it sounds like a misconfigure.

Reply:
Ah ok, I imagined that `requires_sliding_sync` would make sure you build a client with support for the type of sliding sync you configured it with but I see the confusion there. For now I'll revert this change and we can think about it when SSS is a bit more mature.